agent/taskresource/logrouter/logrouter_unix_test.go

```diff
@@ -260,3 +263,22 @@ func TestCleanupLogRouterResourceError(t *testing.T) {
 
 	assert.Error(t, logRouterResource.Cleanup())
 }
+
+func TestInitializeLogRouterResource(t *testing.T) {
+	logRouterResource := newMockLogRouterResource(LogRouterTypeFluentd, testFluentdOptions, nil, nil)
+	logRouterResource.Initialize(&taskresource.ResourceFields{}, status.TaskRunning, status.TaskRunning)
+
+	assert.NotNil(t, logRouterResource.statusToTransitions)
+	assert.Equal(t, 1, len(logRouterResource.statusToTransitions))
+	assert.NotNil(t, logRouterResource.os)
+	assert.NotNil(t, logRouterResource.ioutil)
+}
+
+func TestSetKnownStatus(t *testing.T) {
```

Comment:
nit: would be better to add a test case to make sure when appliedStatusUnsafe > knownState, the applied status not get reset to none.

Reply:
added the test